src/main/java/mcp/mobius/waila/config/JsonConfig.java

```diff
@@ -53,48 +53,55 @@
                 }
                 config = factory.get();
             } else {
-                try (BufferedReader reader = Files.newBufferedReader(this.path, StandardCharsets.UTF_8)) {
-                    config = this.gson.fromJson(reader, clazz);
-                    int version = versionGetter.applyAsInt(config);
-                    if (version != currentVersion) {
-                        Path old = Paths.get(this.path + "_old");
-                        Waila.LOGGER.warn("Config file "
-                            + this.path
-                            + " contains different version ("
-                            + version
-                            + ") than required version ("
-                            + currentVersion
-                            + "), this config will be reset. Old config will be placed at "
-                            + old);
-                        Files.deleteIfExists(old);
-                        Files.copy(this.path, old);
-                        config = factory.get();
-                    } else {
-                        init = false;
-                    }
-                } catch (Exception e) {
-                    Path old = Paths.get(this.path + "_old");
-                    Waila.LOGGER.error("Exception when reading config file "
-                        + this.path
-                        + ", this config will be reset. Old config will be placed at "
-                        + old, e);
-                    try {
-                        Files.deleteIfExists(old);
-                        Files.copy(this.path, old);
-                    } catch (IOException e1) {
-                        Waila.LOGGER.error("well this is embarrassing...", e1);
-                    }
-                    config = factory.get();
-                }
+
+                config = ConfigFileChecker(clazz, versionGetter, currentVersion, factory, initializer);
```

Comment:
Do you even have experience with Java? I don't think you are, passing a variable to a method and modifying it _inside_ the method won't change the original value.